feature (ch05_p01): add easy-to-understand solution

ryoyama0508 opened the pull request:
Added a solution `chapter_05/p01_insertion.py` since the original solution is a bit hard to read and understand for me. I hope this solution will be helpful for people in the same situation.

By the way, I personally think we should change the test cases because the question in the book says _"You are given two 32-bit __numbers__, N and M, and two bit positions, i and j. (question continues)"_. So I think the input of the test case should be __numbers(int)__, not string. My understanding might be weird or wrong since I'm not native speaker.

ryoyama0508 commented:
I changed my code since flake8 doesn't let me use `.format`.

brycedrennan commented on chapter_05/p01_insertion.py:
```diff
@@ -25,9 +25,26 @@ def bits_insertion(n, m, i, j):
+        if (int_m >> idx) & 1 != 0:
+            int_n |= 1 << (idx + i)
+
+    return bin(int_n)[2:]
```
f-strings are the replacement for `.format` 

So you could do:
```python
 return f"{int_n:b}"
```

brycedrennan commented on chapter_05/p01_insertion.py:
```diff
@@ -25,9 +25,26 @@ def bits_insertion(n, m, i, j):
 
 
+def bits_insertion_easy_to_understand(n, m, i, j):
+    # change str into int in order to do bit manipulation
```
I agree that this should just be starting with integers.

brycedrennan commented:
Thanks! I just had some minor suggestions

ryoyama0508 commented:
Thanks for the suggestion. Changed the output to f-string. And also changed the input from string to integers. I wasn't sure what should I do to the other solution, so I just left it as it is.

brycedrennan reviewed (changes requested):
Thanks for those updates. 

It's possible I'm mistaken but it seems this solution does not work if n is set to `int("11111111111", 2)` - the function outputs `11111111111`

Let me know what you think

ryoyama0508 commented:
> It's possible I'm mistaken but it seems this solution does not work if n is set to `int("11111111111", 2)` - the function outputs `11111111111`
> 
> Let me know what you think

I think you're completely right. I fixed that part.

Also, thanks for cleaning the test cases! Appreciate it.

brycedrennan commented:
Thanks - looks good!
